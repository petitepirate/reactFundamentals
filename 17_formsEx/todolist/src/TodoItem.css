button {
	display: inline-block;
}

p {
	display: inline-block;
	margin: 10px 20px;
}

.complete {
	text-decoration: line-through;
}
