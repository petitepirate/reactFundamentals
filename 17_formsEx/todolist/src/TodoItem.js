import React from 'react';
import './TodoItem.css';

function TodoItem( { item = {todo: "Example Todo", isComplete: false}, idx = 1, completeTodo, deleteTodo } ) {
  return (
    <div className="todo-item" id={idx}>
      <button type="button" className="complete-btn" onClick={completeTodo}>{item.isComplete ? 'Mark Incomplete' : 'Mark Complete'}</button>
      <p className={item.isComplete ? 'complete' : 'incomplete'}>{item.todo}</p>
      <button type="button" className="delete-btn" onClick={deleteTodo}>Remove X</button>
    </div>
  )
}

export default TodoItem;
