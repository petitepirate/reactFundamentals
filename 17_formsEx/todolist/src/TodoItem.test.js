import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoItem from './TodoItem';

it('renders without crashing', () => {
  render(<TodoItem />);
});

it('matches snapshot', () => {
  const { asFragment } = render(<TodoItem />)
  expect(asFragment()).toMatchSnapshot();
})

it('runs the complete function on button click', () => {
  const completeMock = jest.fn();
  const { getByText } = render(<TodoItem completeTodo={completeMock} />);
  const completeButton = getByText("Mark Complete");
  fireEvent.click(completeButton);
  expect(completeMock).toHaveBeenCalled();
})

it("runs the delete function on button click", function() {
  const deleteMock = jest.fn();
  const { getByText } = render(<TodoItem deleteTodo={deleteMock} />);
  const deleteButton = getByText("Remove X");
  fireEvent.click(deleteButton);
  expect(deleteMock).toHaveBeenCalled();
});
