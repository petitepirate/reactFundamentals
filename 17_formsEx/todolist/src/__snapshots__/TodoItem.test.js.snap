// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todo-item"
    id="1"
  >
    <button
      class="complete-btn"
      type="button"
    >
      Mark Complete
    </button>
    <p
      class="incomplete"
    >
      Example Todo
    </p>
    <button
      class="delete-btn"
      type="button"
    >
      Remove X
    </button>
  </div>
</DocumentFragment>
`;
